### PR TITLE
fh add: try to match existing `inputs` styles

### PR DESCRIPTION
**Pull request by cole-h**
Rather than inserting a flattened `inputs.nixpkgs.url = "";` above an existing `inputs = { <...> }` structure, we can also insert it inside of that as `inputs = { nixpkgs.url = ""; };`

---

Closes https://github.com/DeterminateSystems/fh/issues/14.

### CODE REVIEW

File: src/cli/cmd/add/flake.rs [hunk 1]
```diff
@@ -148,23 +148,35 @@ impl AttrType {
...
+                    }
 
-                AttrType::insert_input_above_span(
-                    from_span,
-                    flake_contents,
-                    &flake_input,
-                    added_cosmetic_newline,
-                )
+                    // I don't think this is possible, but just in case.
+                    len => Err(color_eyre::eyre::eyre!(
```
Comment (Hoverbear) [inline src/cli/cmd/add/flake.rs:176]: May be wise to emit a warning asking for a bug report, then default to creating the input using our "default" style to get it working for the user?

**Review by Hoverbear — APPROVED** [review]

File: src/cli/cmd/add/flake.rs (hunk 1 above)
Comment (grahamc) [reply src/cli/cmd/add/flake.rs:176]: Nice idea.

**Review by grahamc — APPROVED** [review]
I tested it with a flake with no inputs, a flake with input.foo.url ... format, a flake with input ={ ... format, and a flake with both -- and it behaved exactly like I'd expect in all cases. Nicely done.

File: src/cli/cmd/add/flake.rs (hunk 1 above)
Comment (cole-h) [reply src/cli/cmd/add/flake.rs:176]: Great idea, even; I'll make this part of the "setup tracing" ticket though :+1:

**Pull request merged**